src/lib/support/Span.h

```diff
@@ -390,4 +390,18 @@ inline CHIP_ERROR CopyCharSpanToMutableCharSpan(CharSpan cspan_to_copy, MutableC
     return CHIP_NO_ERROR;
 }
 
+/**
+ * Copies a CharSpan into a MutableCharSpan.
+ * If the span_to_copy does not if in out_span, span_to_copy in truncated to fit in out_span.
+ * @param span_to_copy The CharSpan to copy.
+ * @param out_span The MutableCharSpan in which span_to_copy is to be copied.
+ */
+inline void CopyCharSpanToMutableCharSpanTruncate(CharSpan span_to_copy, MutableCharSpan & out_span)
```

Comment:
I don't feel strongly about this (because both names are terrible and I can't think of a non-terrible one), but maybe:
```suggestion
inline void CopyCharSpanToMutableCharSpanWithTruncation(CharSpan span_to_copy, MutableCharSpan & out_span)
```